### PR TITLE
Translate .Load() to imageLoad() for Vulkan

### PR DESCRIPTION
**Pull request by tfoleyNV**
We were already emitting calls to `imageLoad()` and `imageStore` when a `RWTexture*` was used with `operator[]`:

```hlsl
RWTexture2D<float> myTex;
...
float value = myTex[xy]; // becomes an imageLoad
myTex[xy] = value;       // becomes an imageStore
```

However, we were *not* correctly handling the translation of an explicit `.Load()` operation:

```hlsl
float value = myTex.Load(xy);
```

The `.Load()` operation was being translated to a GLSL `texelFetch` as it would be a for a `Texture2D`, and not to an `imageLoad()` as would make sense for a `RWTexture2D` (which becomes a GLSL `image2D`).

This fix is confined to the stdlib, and is mostly a matter of emitting either `texelFetch` or `imageLoad` as the GLSL function name depending on the "access" of the resource type. It is messy code, but straightforward.

One extra detail was that there had been logic to emit a `, 0` argument in the `texelFetch` calls in the non-read-only case, because `texelFetch` usualy requires an explicit mip-level argument and `.Load()` on a `RWTexture*` doesn't recieve an LOD parameter. This is a non-issue now that we are calling `imageLoad()` instead, because `imageLoad` doesn't need/want the extra argument.

### CODE REVIEW

File: source/slang/hlsl.meta.slang [hunk 1]
```diff
@@ -1439,16 +1440,18 @@ for (int aa = 0; aa < kBaseBufferAccessLevelCount; ++aa)
...
 
     sb << "T Load(int location, out uint status);\n";
 
     sb << "__subscript(uint index) -> T {\n";
 
     sb << "__glsl_extension(GL_EXT_samplerless_texture_functions)";
-    sb << "__target_intrinsic(glsl, \"texelFetch($0, int($1))$z\") get;\n";
+    sb << "__target_intrinsic(glsl, \"" << glslLoadFuncName << "($0, int($1))$z\") get;\n";
 
     if (kBaseBufferAccessLevels[aa].access != SLANG_RESOURCE_ACCESS_READ)
```
Comment (jsmall-nvidia) [inline source/slang/hlsl.meta.slang]: Can just use 'access' variable.

File: source/slang/hlsl.meta.slang.h [hunk 2]
```diff
@@ -1515,16 +1516,18 @@ for (int aa = 0; aa < kBaseBufferAccessLevelCount; ++aa)
...
 
     sb << "T Load(int location, out uint status);\n";
 
     sb << "__subscript(uint index) -> T {\n";
 
     sb << "__glsl_extension(GL_EXT_samplerless_texture_functions)";
-    sb << "__target_intrinsic(glsl, \"texelFetch($0, int($1))$z\") get;\n";
+    sb << "__target_intrinsic(glsl, \"" << glslLoadFuncName << "($0, int($1))$z\") get;\n";
 
     if (kBaseBufferAccessLevels[aa].access != SLANG_RESOURCE_ACCESS_READ)
```
Comment (jsmall-nvidia) [inline source/slang/hlsl.meta.slang.h]: Similarly can just use 'access'.

**Comment by tfoleyNV** [comment]
Strange. The tests passed locally for me. Perhaps I failed to push some of my local changes.

**Comment by tfoleyNV** [comment]
Oh, I see. The test failure is because the baseline GLSL needs to change once I have merged in your recent fixes for `SV_*` semantics translation.

**Pull request merged**